### ISSUE TITLE
SLM2 Build Progress Tracker

### ISSUE THREAD

**Issue by sfreytag** [sfreytag/simonlikesmaps#50]
# SLM2 Initial Release

Last update 15 Dec 2024

## Now Ready!

Update: the initial release is now ready to try - https://new.simonlikesmaps.com (Jan 2024)

For ongoing development, scroll down the following...

## Initial Build (Complete)

This tracks the things that have been done on SLM2 to take it to a first release.

It won't necessarily reach feature parity with SLM1 before a release, but will get close. Any missing features can be added after an initial release.

**Rough task list for first release of SLM2**

- [x] New backend server and database
- [x] New frontend application setup
- [x] New UI layout
- [x] Log in and out
- [x] Create a plan
- [x] Edit a plan
    - [x] Cancel edits and revert to original
    - [x] Auto show settings modal for everything but first route 
- [x] Delete a plan
- [x] Plan tagging
- [x] Add lines to a plan
- [x] Add routes to a plan
- [x] Add waypoints to a plan
- [x] Support many layers for map
- [x] Support many layers for overlays
- [x] Migrate all the maps
- [x] Migrate all the overlays
- [x] Tools: Satellite spy
- [x] Tools: Your Location
- [x] Tools: Zoom in and out
- [x] Elevation profiles
    - [x] View the profile and use it to interact with the map
    - [x] Profile stats (ascent, descent, smoothing, sample interval, gradient) 
    - [x] Colour lines for gradients
- [x] Ability to migrate routes from SLM1
    - [x] Build API on old SLM
    - [x] Connect to old SLM to get old routes
    - [x] Convert old routes to new plans and save
- [x] Port initial help/about content
- [x] Searching in PlanView (by name and tags)
- [x] Planning tools
    - [x] Default map 
    - [x] Feature (line/route/waypoint) colours
    - [x] Validate on plan save data
- [x] Overlay opacity
- [x] Attributions control and audit the attributions
- [x] Map scale

**Bugs**

- [x] Throttle/debounce line drag when editing
- [x] Elevation profile width when planning but no menu open
- [x] Elevation profile tracker marker should be in same colour as route and circular
- [x] Load user view only once rather than via watch, and before map loads
- [x] Be able to view elevation while editing
- [x] Check current map is selected option when saving new plan
- [x] Ensure map is loaded before displaying & fitting to route (in the case where the app loads on non-map view, eg your plans view)
- [x] Elevation profile width (the tools overflow)
- [x] Delete route points on mobile
- [x] Routes index should not include GeoJSON
- [x] Elevation:
    - [x] Labels space when elev > 1000m
    - [x] X-axis rounding on long routes
    - [x] elev prog marker getting added to plan edit
- [x] PlanEditor spacing when length > 1000km
- [x] Change a line to a route and vice-versa (if easy)

## SLM2 Feature Parity (In Progress)

Here's the list of upcoming features that will get added to the new version of SimonLikesMaps. The goal here is to get to the same level of functionality as the old SLM so that the old one can then be turned off. They are roughly in priority order.

- [x] Search for placenames
- [x] GPX import
- [x] GPX export
- [x] Plan sharing
    - [x] Share an individual plan with a link 
    - [x] View plans previously shared with you in 'Your Plans'
    - [x] When deleting a route, delete any linked shares - see #28 
    - [x] Ensure old-style links from SLMv1 still work?
- [x] Port remaining help content
    - [x] Link help to maps / overlays
- [x] Tools: Location picker --> bumped to #63
- [x] Tools: Reticle --> bumped to #62 
- [x] JWT renew tokens to persist session past 30 days --> bumped to #65  (not required for parity)
- [x] Tools: Print screen --> bumped to #64
- [x] Plan/Route printing --> bumped to #64

## Bugs

- [x] Clone plan model for plan settings modal
- [x] Manage map visibility at extremes --> #66 (was already an issue with SLMv1)
- [x] Strip cookies from GetCapabilities requests
- [x] Audit elevation calcs - seem quite high
- [x] PlanEditor vertical scrolling broken in iOS/Safari
- [x] Removing tags when saving plan - works, but UI does not keep up
- [x] Check emails are lower case
- [x] Error is a modal, then error message modal appears behind not in front
- [x] Hide a route before deleting it
- [x] MapBanner at top of stack

## Feedback

- [x] Make entire search row clickable
- [x] View feature settings automatically when first created (not not thereafter unless requested)
- [x] Less vertical space for waypoints and lines (without distance, so not needed)

## Migration

- [x] Migrate www.slm
- [x] Migrate new.slm
- [x] Switch DNS host to support cname flattening
- [x] Set up redirect for new to www in new host
- [x] Wait 72hrs for all internet to catch up then remove new subdomain from server (rely on DNS instead)
- [x] Add a CNAME root record

**Comment by sfreytag** [comment]
31 Jul - ticked off 'edit a plan' and 'Tools: Location picker'

**Comment by sfreytag** [comment]
25th October - Ticked off a couple of small things and tagged the remainder [in progress] or [not started]

Currently working on ElevationProfiles and migrating old routes.

**Comment by sfreytag** [comment]
26th October - Refining the list of what might count as V1

**Comment by sfreytag** [comment]
29th October - all overlays ported

**Comment by sfreytag** [comment]
3rd Nov - Ticked off overlay opacity

**Comment by sfreytag** [comment]
**Rough task list of things that might come after the first release**

- [ ] GPX import / export
- [ ] Tools: Print screen
- [ ] Plan/Route printing
- [ ] Plan sharing
- [ ] Mobile app and offline routes
- [ ] Search for placenames
- [ ] Tools: Location picker

**Comment by tasker25** [comment]
Problem raised in #49 is solved in new version, so no need for enhancement to strip extraneous tags from uploaded track files.

**Comment by sfreytag** [comment]
Mobile app being tracked in #37

**Comment by sfreytag** [comment]
Some features required for parity are now bumped to #62, #63, #64, #65 & #66

This will allow me to get closer to managing a 'finished' SLMv2 such that I can finally turn off the old one, and move the domain names around such that new.simonlikesmaps.com takes over www.simonlikesmaps.com

**Comment by sfreytag** [comment]
Migration ongoing today

**Comment by sfreytag** [comment]
SLM2 is now running at https://simonlikesmaps.com and https://www.simonlikesmaps.com

https://new.simonlikesmaps.com is redirecting to www

Therefore the build of SLM2 is done. Things to tidy up (turn off the old one, remove code used during migrate) is tracked in #69
